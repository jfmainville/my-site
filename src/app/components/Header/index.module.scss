@import "../../variables.module.scss";

.Header {
  display: flex;
  position: relative;
}

.Navbar {
  display: flex;
  flex-direction: row-reverse;
  position: absolute;
  padding: 2rem;
  right: 0;
}

.Link {
  font-size: 1.17em;
  color: $text-color;
  padding: 0 1rem 0 1rem;
}

.TextHeading {
  margin: 8rem 0 0 2rem;
  color: $text-color;
}
