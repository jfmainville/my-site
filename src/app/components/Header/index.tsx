import Link from "next/link";
import styles from "./index.module.scss";
import Image from "next/image";

const Header = () => {
  return (
    <div id={"home"} className={styles.Header}>
      <Image
        src="/images/bromont.png"
        width={0}
        height={0}
        sizes="100vw"
        alt=""
        style={{ width: "100%", height: "100vh" }}
      />
      <div className={styles.Navbar}>
        <Link className={styles.Link} href={"/#contact"}>
          Contact
        </Link>
        <Link hidden={true} className={styles.Link} href={"/#blog"}>
          Blog
        </Link>
        <Link className={styles.Link} href={"/#experience"}>
          Experience
        </Link>
        <Link className={styles.Link} href={"/#about"}>
          About
        </Link>
      </div>
      <h1 className={styles.TextHeading} style={{ position: "absolute" }}>
        Hey, I&apos;m Jean-Frederic, welcome to my site! ðŸ”¥
      </h1>
    </div>
  );
};

export default Header;
