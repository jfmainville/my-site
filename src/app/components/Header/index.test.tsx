import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "./index";

describe("<Header/> Test Suite", () => {
  it("should render the <Header/> component unchanged", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it("should show the header text in the <Header /> component", () => {
    render(<Header />);
    const headerText = screen.getByText(
      "Hey, I'm Jean-Frederic, welcome to my site! ðŸ”¥",
    );
    expect(headerText).toBeInTheDocument();
  });

  it("should show the background picture in the <Header /> component", () => {
    render(<Header />);
    const backgroundPicture = screen.getByRole("presentation");
    expect(backgroundPicture).toBeInTheDocument();
  });
});
