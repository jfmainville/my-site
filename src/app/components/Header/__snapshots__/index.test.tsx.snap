// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Test Suite should render the <Header/> component unchanged 1`] = `
<div>
  <div
    class="Header"
    id="home"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="0"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fbromont.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Fbromont.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fbromont.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fbromont.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fbromont.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fbromont.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fbromont.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fbromont.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fbromont.png&w=3840&q=75 3840w"
      style="color: transparent; width: 100%; height: 100vh;"
      width="0"
    />
    <div
      class="Navbar"
    >
      <a
        class="Link"
        href="/#contact"
      >
        Contact
      </a>
      <a
        class="Link"
        hidden=""
        href="/#blog"
      >
        Blog
      </a>
      <a
        class="Link"
        href="/#experience"
      >
        Experience
      </a>
      <a
        class="Link"
        href="/#about"
      >
        About
      </a>
    </div>
    <h1
      class="TextHeading"
      style="position: absolute;"
    >
      Hey, I'm Jean-Frederic, welcome to my site! ðŸ”¥
    </h1>
  </div>
</div>
`;

exports[`<Header/> Test Suite should render the <Header/> component unchanged 2`] = `
<div>
  <div
    class="Header"
    id="home"
  >
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="0"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fimages%2Fbromont.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Fbromont.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fbromont.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fbromont.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fbromont.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fbromont.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fbromont.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fbromont.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fbromont.png&w=3840&q=75 3840w"
      style="color: transparent; width: 100%; height: 100vh;"
      width="0"
    />
    <div
      class="Navbar"
    >
      <a
        class="Link"
        href="/#contact"
      >
        Contact
      </a>
      <a
        class="Link"
        hidden=""
        href="/#blog"
      >
        Blog
      </a>
      <a
        class="Link"
        href="/#experience"
      >
        Experience
      </a>
      <a
        class="Link"
        href="/#about"
      >
        About
      </a>
    </div>
    <h1
      class="TextHeading"
      style="position: absolute;"
    >
      Hey, I'm Jean-Frederic, welcome to my site! ðŸ”¥
    </h1>
  </div>
</div>
`;
